rust/main/agents/relayer/src/msg/op_submitter.rs

```diff
@@ -342,86 +371,247 @@ async fn prepare_task(
     max_submit_queue_len: Option<u32>,
     metrics: SerialSubmitterMetrics,
 ) {
-    // Prepare at most `max_batch_size` ops at a time to avoid getting rate-limited
-    let ops_to_prepare = max_batch_size as usize;
     loop {
-        // Apply backpressure to the prepare queue if the submit queue is too long.
-        if let Some(max_len) = max_submit_queue_len {
-            let submit_queue_len = submit_queue.len().await as u32;
-            if submit_queue_len >= max_len {
-                debug!(
-                    %submit_queue_len,
-                    max_submit_queue_len=%max_len,
-                    "Submit queue is too long, waiting to prepare more ops"
-                );
-                // The submit queue is too long, so give some time before checking again
-                sleep(Duration::from_millis(150)).await;
-                continue;
-            }
+        if apply_backpressure(&submit_queue, &max_submit_queue_len).await {
+            continue;
```

Comment:
the name `apply_backpressure` indicates sleeping isn't done inside the function. Maybe move out the sleep from there in here, before `continue`ing?

Reply:
Done